### ISSUE TITLE
Rounding issues differing from ingame

### ISSUE THREAD

**Issue by Mamotromico** [Openarl/PathOfBuilding#27]
There seems to be something different from how the app calculate numbers and the game shows us. Example:

![image](https://cloud.githubusercontent.com/assets/6295903/18854859/b4257246-8424-11e6-9a05-9c4abf1b2837.png)

![image](https://cloud.githubusercontent.com/assets/6295903/18854885/da2e1c9a-8424-11e6-8441-2034cd53ccd9.png)

Same itens, tree, etc, perfectly replicated etc. On this character the DPS calculation is off by just .3, but on a higher level character this difference was about 3K DPS (from 32K to 35K)

You could check how exactly the game calculates the numbers so you can guarantee parity on this

But really, amazing tool. I'm loving it, and the copy-paste support when adding item is something awesome

**Comment by Openarl** [comment]
Unfortunately I've never been able to find any information regarding the exact rounding methods used in the game's calculations. Through trial and error I'd already managed to sort out the rounding for the defence stats (life/mana/armour etc) but the damage calculations are vastly more complex. It's something I'd like to fix some day, but it's not a high priority as it should only cause errors in the order of 1%. A difference of 3K out of 35K however is way beyond rounding error, so there must be something else going on in that case. I'd need more details to be able to determine what that might be, of course.

**Comment by Mamotromico** [comment]
Yeah I'll redo the other character later today and check it again, and I'll post results here, it might have been my mistake.

I'll keep you updated and see if I can find anything on this issue

**Comment by myrahz** [comment]
I'll try to see where I can help on the rounding calculation, I've had that problem myself and debugged it exaustively, I'll let you know once I arrive home.

**Comment by Openarl** [comment]
Oh, I've done my fair share of debugging on this. I think I've got it nailed down, but I'd need to make quite a few modifications to the hit damage calculation code to actually implement it, which is the main reason I haven't done so yet.
